service/src/main/java/bio/terra/tanagra/service/artifact/CohortService.java

```diff
@@ -125,6 +129,49 @@ public Cohort updateCohort(
     return cohortDao.getCohort(cohortId);
   }
 
+  /** Clone an existing cohort's revisions, including reviews and annotations */
+  public Cohort cloneCohort(
+      String studyId,
+      String cohortId,
+      String userEmail,
+      String destinationStudyId,
+      @Nullable String displayName,
+      @Nullable String description) {
+    Cohort original = getCohort(studyId, cohortId);
+
+    String newDisplayName = displayName;
+    if (newDisplayName == null && original.getDisplayName() != null) {
+      newDisplayName = original.getDisplayName();
+      if (studyId.equals(destinationStudyId)) {
+        newDisplayName = "Copy of: " + newDisplayName;
+      }
+    }
+    String newDescription = description;
+    if (newDescription == null && original.getDescription() != null) {
+      newDescription = original.getDescription();
+      if (studyId.equals(destinationStudyId)) {
+        newDescription = "Copy of: " + newDescription;
```

Comment:
Don't append this to the description. The name is enough.

Reply:
sgtm, done